server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -113,28 +123,42 @@ export function verifyStorageToken(
             return res.status(403).send("Missing tenantId in request.");
         }
         const documentId = getParam(request.params, "id") || request.body.id;
-        if (requireDocumentId && !documentId) {
+        if (options.requireDocumentId && !documentId) {
             return res.status(403).send("Missing documentId in request");
         }
         let claims: ITokenClaims;
+        let tokenLifetimeMs: number | undefined;
         try {
-            claims = validateTokenClaims(token, documentId, tenantId, requireDocumentId);
+            claims = validateTokenClaims(token, documentId, tenantId, options.requireDocumentId);
             if (isTokenExpiryEnabled) {
-                validateTokenClaimsExpiration(claims, maxTokenLifetimeSec);
+                tokenLifetimeMs = validateTokenClaimsExpiration(claims, maxTokenLifetimeSec);
             }
         } catch (error) {
             if (error instanceof NetworkError) {
                 return res.status(error.code).send(error.message);
             }
             throw error;
         }
-        tenantManager.verifyToken(claims.tenantId, token)
-            .then(() => {
-                next();
-            })
-            .catch((error) => {
-                return res.status(403).json(error);
-            });
+        try {
+            await tenantManager.verifyToken(claims.tenantId, token);
+        } catch (error) {
+            return res.status(403).json(error);
+        }
+
+        if (options.ensureSingleUseToken) {
+            // TODO: remove `as any` after #7065 is merged and released
+            const singleUseKey = (claims as any).jti ?? token;
+            // TODO: should we eat cache errors here or block flow?
+            if (await options.singleUseTokenCache?.get(singleUseKey).catch(() => false)) {
```

Comment:
Curious, does this syntax mean that if the `get()` call generates an exception, we will just ignore it and return false, so that the if condition is not met?
Otherwise, the token already exists in the cache, meaning it has already been used, so we should fail the request?

Reply:
Correct, yeah. Not sure if this is the desired flow, hence the TODO statement above. @tanviraumi could you weigh in?

Reply:
Yeah thats the correct approach. We don't want to block creation because of redis failure.